hudi-io/src/test/java/org/apache/hudi/io/storage/BaseTestStorageConfiguration.java

```diff
@@ -71,13 +72,25 @@ public abstract class BaseTestStorageConfiguration<T> {
 
   @Test
   public void testConstructorNewInstanceUnwrapCopy() {
-    T conf = getConf(EMPTY_MAP);
+    T conf = getConf(prepareConfigs());
     StorageConfiguration<T> storageConf = getStorageConfiguration(conf);
     StorageConfiguration<T> newStorageConf = storageConf.newInstance();
+    Class unwrapperConfClass = storageConf.unwrap().getClass();
     assertNotSame(storageConf, newStorageConf);
+    validateConfigs(newStorageConf);
     assertNotSame(storageConf.unwrap(), newStorageConf.unwrap());
     assertSame(storageConf.unwrap(), storageConf.unwrap());
+    assertSame(storageConf.unwrap(), storageConf.unwrapAs(unwrapperConfClass));
     assertNotSame(storageConf.unwrap(), storageConf.unwrapCopy());
+    validateConfigs(getStorageConfiguration(storageConf.unwrapCopy()));
+    assertNotSame(storageConf.unwrap(), storageConf.unwrapCopyAs(unwrapperConfClass));
+    validateConfigs(getStorageConfiguration((T) storageConf.unwrapCopyAs(unwrapperConfClass)));
```

Comment:
Can we add some comments in the assertions.

Reply:
Addressed.